### DISCUSSION TITLE
Rule apparently not triggering but works when running `--apply`

### DISCUSSION THREAD

**Discussion by abey79** [asmvik/yabai#2297] [Q&A]
Hello all, new user here :wave:

I'm getting confusing behaviour with the RustRover IDE and rules. I'm using a separate window for the diff view (e.g. when reviewing a commit or handling merge conflict), and I'd like to have it floating.

<img width="1167" alt="image" src="https://github.com/koekeishiya/yabai/assets/49431240/dd03ff27-19e0-4f95-ba67-4a89bf1b39e9">
<details>
<summary>Query output for that window</summary>

```json
{
        "id":24828,
        "pid":63340,
        "app":"RustRover",
        "title":"Commit: main.rs",
        "scratchpad":"",
        "frame":{
                "x":-1796.0000,
                "y":-186.0000,
                "w":1792.0000,
                "h":788.0000
        },
        "role":"AXWindow",
        "subrole":"AXStandardWindow",
        "root-window":true,
        "display":2,
        "space":6,
        "level":0,
        "sub-level":0,
        "layer":"normal",
        "sub-layer":"normal",
        "opacity":1.0000,
        "split-type":"horizontal",
        "split-child":"first_child",
        "stack-index":0,
        "can-move":true,
        "can-resize":true,
        "has-focus":false,
        "has-shadow":true,
        "has-parent-zoom":false,
        "has-fullscreen-zoom":false,
        "has-ax-reference":true,
        "is-native-fullscreen":false,
        "is-visible":true,
        "is-minimized":false,
        "is-hidden":false,
        "is-floating":false,
        "is-sticky":false,
        "is-grabbed":false
}
```

</details>

I have a rule set for that:
```
yabai -m rule --add label="RR Commit" app='^RustRover$' title='^Commit.*' manage=off
```

However, that rule seems to no proc when the commit window is opened. I originally thought something was wrong with the rule itself, but if I then `--apply` it, the commit window does become floating at that moment:

```
yabai -m rule --apply
```

Am I missing something obvious? Or could this be a weird behaviour specific to RustRover?

**Comment by abey79** [comment]
Mm it may be an instance of https://github.com/koekeishiya/yabai/issues/1687

TLDR: JetBrain IDEs apparently spawn windows without a title and subsequently rename them. This explains why the rule don't proc.

**Reply by abey79** [reply]
I've tried this as a workaround:
```
yabai -m signal --add event=window_title_changed app='^RustRover$' action="yabai -m rule --apply"
```

It works, but the window is still first resized, then set as floating. The initial resize is really annoying, so not really a possibility.

**Reply by abey79** [reply]
OK, I think I have a working solution:

```
# make sure any window is managed - necessary when running `--apply`
yabai -m rule --add label="RR named" app='^RustRover$' manage=on

# any unnamed window shouldn't be managed
yabai -m rule --add label="RR unnamed" app='^RustRover$' title='^$' manage=off

# regular rules
yabai -m rule --add label="RR Merge" app='^RustRover$' title='^Merge Revisions.*' manage=off
yabai -m rule --add label="RR Commit" app='^RustRover$' title='^Commit.*' manage=off
yabai -m rule --add label="RR Goto" app='^RustRover$' title='^Go to Line:Column$' manage=off
yabai -m rule --add label="RR Settings" app='^RustRover$' title='^Settings$' manage=off
#...probably more

# if any window is renamed, reapply the rules
yabai -m signal --add event=window_title_changed label="RustRover" app='^RustRover$' action="yabai -m rule --apply"
```
